forEach.js

```diff
@@ -11,3 +11,44 @@ function cb(item, index, arr){
 };
 
 customForEach(family, cb);
+
+// Slice
+const array = ["t", "e", "s", "r"];
+
+function slice(arr,start,end) {
+  end = end || arr.length;
+  const newArr = [];
+  for(let i = start; i < arr.length; i++){
+    newArr.push(arr[i]);
+    if (arr[i] === arr[end - 1]) return newArr ;
+  }
```

Comment:
Instead you can loop while the i is less then the end and just push the result
```suggestion
  for(let i = start; i < finish; i++){
    newArr.push(arr[i]);
  }
  return newArr;
```